### PR TITLE
sync test

### PR DESCRIPTION
**Pull request by samfrb**

### CODE REVIEW

File: AElf.OS.Tests/ForkDownloadJobTest.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        public ForkDownloadJobTest()
+        {
+            _blockChainService = GetRequiredService<IBlockchainService>();
+            _peerPool = GetRequiredService<IPeerPool>();
+            _blockExecService = GetRequiredService<IBlockchainExecutingService>();
+            _netService = GetRequiredService<INetworkService>();
+            
+            // Build download job because it's difficult to injection for this. 
+            _job = new ForkDownloadJob();
```
Comment (loning) [inline AElf.OS.Tests/ForkDownloadJobTest.cs]: just inject the job

File: AElf.OS.Tests/Network/NetTestAElfModule.cs [hunk 2]
```diff
@@ -1,13 +1,99 @@
...
+using Moq;
+using NSubstitute;
+using Volo.Abp;
 using Volo.Abp.EventBus;
 using Volo.Abp.Modularity;
 
 namespace AElf.OS.Tests.Network
 {
-    [DependsOn(typeof(TestBaseAElfModule), typeof(AbpEventBusModule))]
+    [DependsOn(typeof(TestBaseAElfModule), typeof(AbpEventBusModule), typeof(TestsOSAElfModule), typeof(TestBaseKernelAElfModule))]
```
Comment (loning) [inline AElf.OS.Tests/Network/NetTestAElfModule.cs]: AbpEventBusModule remove

File: AElf.OS.Tests/TestsOSAElfModule.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 namespace AElf.OS.Tests
 {
-    [DependsOn(typeof(TestBaseAElfModule)), DependsOn(typeof(AbpEventBusModule)), DependsOn(typeof(AbpBackgroundJobsModule))]
+    [DependsOn(typeof(TestBaseAElfModule)), DependsOn(typeof(CoreOSAElfModule)), DependsOn(typeof(AbpEventBusModule)), DependsOn(typeof(AbpBackgroundJobsModule))]
```
Comment (loning) [inline AElf.OS.Tests/TestsOSAElfModule.cs]: remove AbpBackgroundJobsModule

File: AElf.OS.Tests/Network/NetTestAElfModule.cs [hunk 4]
```diff
@@ -1,13 +1,99 @@
...
     public class NetTestAElfModule : AElfModule
     {
+        private static int height = 0;
```
Comment (loning) [inline AElf.OS.Tests/Network/NetTestAElfModule.cs]: no static

File: AElf.OS.Tests/Network/NetTestAElfModule.cs [hunk 5]
```diff
@@ -1,13 +1,99 @@
...
+                .Returns(new List<IPeer> { peerMock.Object });
+
+            context.Services.AddSingleton<IPeerPool>(peerPoolMock.Object);
+
+            context.Services.AddTransient<ForkDownloadJob>();
+        }
+        
+        public override void OnPreApplicationInitialization(ApplicationInitializationContext context)
+        {
+            //init test data here
```
Comment (loning) [inline AElf.OS.Tests/Network/NetTestAElfModule.cs]: not here, maybe in KernelTestAElfModule

Comment (samfrb) [reply AElf.OS.Tests/Network/NetTestAElfModule.cs]: causes other tests to fail, will be refactored later.

**Pull request merged**